roc.go

```diff
@@ -0,0 +1,71 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import "sort"
+
+type byY struct {
+	y []float64
+	c []bool
+}
+
+func (a byY) Len() int { return len(a.y) }
+func (a byY) Swap(i, j int) {
+	a.y[i], a.y[j] = a.y[j], a.y[i]
+	a.c[i], a.c[j] = a.c[j], a.c[i]
+}
+func (a byY) Less(i, j int) bool { return a.y[i] < a.y[j] }
+
+// ROC returns paired false positive rate (FPR) and true positive
+// rate (TPR) values which describe the relative (or receiver)
+// operator characteristic (ROC) curve obtained when y is
+// treated as a binary classifier for c.
+// For a given cutoff value, observations corresponding to
+// entries in y greater than the cutoff are classified as
+// true, and these assigned class labels are compared to the
+// true class labels in c to calculate FPR and TPR for all
+// possible cutoffs.
+// More details about ROC curves are available at
+// https://en.wikipedia.org/wiki/Receiver_operating_characteristic
+// Note that ROC will modify both inputs -- sorting them in order
+// of increasing y values.
+func ROC(y []float64, c []bool) (tpr, fpr []float64) {
```

Comment:
>What do you think about making c []float64 instead of bool and specifying that positive values/ negative values indicate the classes and the values indicate weights? So using 1s and -1s would correspond to the unweighted case?

Please don't do this. We use a nil to indicate absence of weighting.